spec/unit/application/ssl_spec.rb

```diff
@@ -0,0 +1,291 @@
+require 'spec_helper'
+require 'puppet/application/ssl'
+require 'webmock/rspec'
+require 'openssl'
+
+describe Puppet::Application::Ssl, unless: Puppet::Util::Platform.jruby? do
+  let(:ssl) { Puppet::Application[:ssl] }
+  let(:name) { 'ssl-client' }
+
+  def generate_cert(name, issuer = nil, issuer_key = nil)
```

Comment:
nit: we added some SSL related helpers in [puppet_spec/ssl](https://github.com/puppetlabs/puppet/blob/master/spec/lib/puppet_spec/ssl.rb) to help with this. Can clean up in a maint commit later.